test/extensions/filters/http/wasm/wasm_filter_test.cc

```diff
@@ -1439,7 +1439,7 @@ TEST_P(WasmHttpFilterTest, RootId2) {
   setupFilter("context2");
   EXPECT_CALL(filter(), log_(spdlog::level::debug, Eq(absl::string_view("onRequestHeaders2 2"))));
   Http::TestRequestHeaderMapImpl request_headers;
-  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter().decodeHeaders(request_headers, true));
+  EXPECT_EQ(Http::FilterHeadersStatus::StopAllIterationAndWatermark, filter().decodeHeaders(request_headers, true));
```

Comment:
this is a cherry pick from https://github.com/envoyproxy/envoy/pull/13930 so we want that PR to be merged first